File: src/RNetPi.Core/Packets/PacketC2SRequestSourceProperties.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Client -> Server
/// ID = 0x33
/// Request Source Properties
/// Data:
///     (Unsigned Char) Source ID
/// </summary>
public class PacketC2SRequestSourceProperties : PacketC2S
{
    public const byte ID = 0x33;

    public byte SourceID { get; private set; }

    public PacketC2SRequestSourceProperties(byte[] data) : base(data)
    {
    }

    public override byte GetID() => ID;

    protected override void ParseData()
    {
        SourceID = Reader.ReadByte();
    }
}

File: src_js/server/packets/PacketC2SRequestSourceProperties.js
const PacketC2S = require("./PacketC2S");

/**
 * Client -> Server
 * ID = 0x33
 * Request Properties
 */
class PacketC2SRequestSourceProperties extends PacketC2S {
    parseData() {
        this._sourceID = this._buffer.readUInt8();
    }

    getSourceID() {
        return this._sourceID;
    }

    getID() {
        return PacketC2SRequestSourceProperties.ID;
    }
}

PacketC2SRequestSourceProperties.ID = 0x33;

module.exports = PacketC2SRequestSourceProperties;
